packages/telegram-cloud-storage-stamper/README.md

```diff
@@ -0,0 +1,195 @@
+- Both an API key and a Cloud Storage key: Store the passed API key at the specified Telegram Cloud Storage key location and set that as the signing key
+  
+The `.getAPIKey()` and `.setAPIKey()` functions operate in a similar manner taking an optional `key` parameter that will be used to `get` or `set` the API key at that location if it is passed, or at the default location if it is not passed.
+
+### Usage
+Insert a new api key into Telegram Cloud Storage at the default API key location
+
+```ts
+import { TelegramCloudStorageStamper } from "@turnkey/telegram-cloud-storage-stamper";
+import { TurnkeyBrowserClient, TurnkeySDKClientConfig } from "@turnkey/sdk-browser";
+
+// create a new Telegram Cloud Storage Stamper
+const stamper = await TelegramCloudStorageStamper.create({
+  apiPublicKey: "...",
+  apiPrivateKey: "...",
+});
+
+if (!stamper) {
+  // There was a problem initializing the stamper
+}
+
+// use the stamper in the client config
+const browserConfig: TurnkeySDKClientConfig = {
+  stamper: stamper,
+  apiBaseUrl: "https://api.turnkey.com",
+  organizationId: <ORGANIZATION_ID>,
+};
+
+// create a TurnkeyClient with the initialized Telegram Cloud Storage Stamper
+const client = new TurnkeyBrowserClient(browserConfig);
+
+// make a request with the client
+const whoamiResponse = await client.getWhoami({
+  organizationId: <ORGANIZATION_ID>,
+});
+```
+
+Use an existing key that has been previously stored in Telegram Cloud Storage at the default API key location key location
+
+```ts
+import { TelegramCloudStorageStamper } from "@turnkey/telegram-cloud-storage-stamper";
+import { TurnkeyBrowserClient, TurnkeySDKClientConfig } from "@turnkey/sdk-browser";
+
+// create a new Telegram Cloud Storage stamper
+const stamper = await TelegramCloudStorageStamper.create();
+
+if (!stamper) {
+  // There was a problem initializing the stamper
+}
+
+// use the stamper in the client config
+const browserConfig: TurnkeySDKClientConfig = {
+  stamper: stamper,
+  apiBaseUrl: "https://api.turnkey.com",
+  organizationId: <ORGANIZATION_ID>,
+};
+
+// create a TurnkeyClient with the initialized Telegram Cloud Storage Stamper
+const client = new TurnkeyBrowserClient(browserConfig);
+
+// make a request with the client
+const whoamiResponse = await client.getWhoami({
+  organizationId: <ORGANIZATION_ID>,
+});
+```
+
+View an entry in Telegram Cloud Storage without inserting an API key, note the usage difference between the `new` and `.create()` here. `.create()` will do the work of getting/setting an API key in Cloud Storage where as  `new` will not
+
+```ts
+import { TelegramCloudStorageStamper } from "@turnkey/telegram-cloud-storage-stamper";
+
+// create a new Telegram Cloud Storage Stamper
+const stamper = new TelegramCloudStorageStamper();
+
+if (!stamper) {
+  // There was a problem initializing the stamper
+}
+
+// the key used to index Telegram Cloud Storage
+const telegramCloudStorageKey = "telegramCloudStorageKey";
```

Comment:
For consistency, we could type this out a tiny bit differently using a `@turnkey/` prefix, similar to what's been defined here: https://github.com/tkhq/sdk/blob/main/packages/sdk-browser/src/storage.ts

Reply:
changed